libr/core/cmd_seek.c

```diff
@@ -138,341 +138,401 @@ R_API int r_core_lines_initcache (RCore *core, ut64 start_addr, ut64 end_addr) {
 	return -1;
 }
 
+static void seek_to_register(RCore *core, const char *input, bool is_silent) {
+	ut64 off;
+	if (core->io->debug) {
+		off = r_debug_reg_get (core->dbg, input);
```

Comment:
I actually just took the original code and moved it in this function. Anyway, I think we can't do what you said because in the else case `r_debug_reg_get` is done after `core->dbg->reg = core->anal->reg`.